### ISSUE TITLE
string quoting in record I/O

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#5052]
### Summary of the proposed change:

Fields in records quote strings:

```
record R {
  var s:string;
}
```

Before this patch,

```
writeln(new R("hi"))
```

would print out

```
(hi)
```

After the patch, it will print out

```
("hi")
```

### Rationale

I'd like to be able to write and then read an array/record/whatever and expect it to work, but there are problems if the object contains strings.  I'm proposing a change to the I/O behavior to fix this problem. 

How did we get here? If you do:

  writeln("hello world\n");

you (naturally?) expect "hello world" to print without quotes:

  hello world

but then if you were reading that input and you did this:

  var s: string;
  readln(s);

`s` would contain only "hello", because the default string reading rule is to read one word only. Even if that were not the case, we wouldn't know when we reached the end of the string since it is not delimited in any way. (and making writeln print the quotes by default interferes with things like writeln("a = ", a) ).

When we get to a record, I don't think these choices are reasonable any more.

For example, if I have

  record R { a: string, b: string }
  writeln(new R("hello world", "second string"));

it will output

  (a = hello world, b = second string)

and there's pretty much no way that the corresponding readln would work (what if the string contains commas or parenthesis?). I believe that the current code will only be able to read 1-word strings in records...

So, I'd like to propose the following change in I/O behavior:

- writeln(string) and readln(string) continue to work the way they do
- write/writeln/read/readln of any aggregate (record/class/array/tuple/etc) will switch the string formatting to double-quoted strings that would work in Chapel source code, so the second record example above would now output

  (a = "hello world", b = "second string")

**Comment by mppf** [comment]
PR #2686 makes this change and was mostly completed before I had to focus on other things. However, there still needs to be some consensus on whether or not to do the change (vs how to implement it).

**Comment by lydia-duncan** [comment]
In the case where a string contains an escaped character (such as `"` or `'`) we should probably print it out like `/"`, right?

**Comment by mppf** [comment]
Yes. Selecting a quoted-string I/O style will cause both the outer quotes and escaping of inner quotes.

**Comment by stonea** [comment]
@mppf, I'm doing some nitpicking of our backlog labeling for issues week.  @mppf You created this issue with the "steal" label but never assigned it to yourself.  Would you mind either assigning yourself this issue or removing the "steal" label?
